src/polycubed/src/utils/netlink.cpp

```diff
@@ -958,5 +961,58 @@ void Netlink::move_iface_into_ns(const std::string &iface, int fd) {
   netlink_nl_send(nlmsg);
 }
 
+std::string Netlink::get_iface_mac(const std::string &iface) {
+  unsigned char mac[IFHWADDRLEN];
+  struct ifreq ifr;
+  int fd, rv;
+
+  strcpy(ifr.ifr_name, iface.c_str());
+  fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
+  if (fd < 0) {
+    throw std::runtime_error(std::string("get_iface_mac error opening socket: ") + std::strerror(errno));
+  } else {
+    rv = ioctl(fd, SIOCGIFHWADDR, &ifr);
+    if (rv >= 0)
+      memcpy(mac, ifr.ifr_hwaddr.sa_data, IFHWADDRLEN);
+    else {
+      close(fd);
+      throw std::runtime_error(std::string("get_iface_mac error determining the MAC address: ") + std::strerror(errno));
+    }
+  }
+  close(fd);
+
+  uint64_t mac_;
+  memcpy(&mac_, mac, sizeof mac_);
+  return polycube::service::utils::be_uint_to_mac_string(mac_);
+}
+
+std::string Netlink::get_iface_ip(const std::string &iface) {
```

Comment:
Did you test this function? It throws `IP Address is not in a valid format` in some experiments I am doing.